examples/nlp_example.py

```diff
@@ -142,6 +142,10 @@ def collate_fn(examples):
                 optimizer.step()
                 lr_scheduler.step()
                 optimizer.zero_grad()
+        
+        accelerator.wait_for_everyone()
+        unwrapped_model = accelerator.unwrap_model(model)
+        unwrapped_model.save_pretrained('accl_out', save_function=accelerator.save, state_dict=accelerator.get_state_dict(model))
```

Comment:
This should be an argument of the script, and we should just save if it is set.

Reply:
Sure tha

Reply:
Do you want me to add the argument as a part of this pr or the next

Reply:
In this PR please :-)